apps/wallet-mobile/src/features/Receive/common/ReceiveProvider.tsx

```diff
@@ -0,0 +1,90 @@
+import {produce} from 'immer'
+import _ from 'lodash'
+import React from 'react'
+
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useReceiveAddresses} from '../../../yoroi-wallets/hooks'
+
+export const useReceive = () => React.useContext(ReceiveContext)
+
+export const ReceiveProvider = ({
+  children,
+  initialState,
+}: {
+  children: React.ReactNode
+  initialState?: Partial<ReceiveState>
+}) => {
+  const [state, dispatch] = React.useReducer(receiveReducer, {
+    ...defaultState,
+    ...initialState,
+  })
+  const wallet = useSelectedWallet()
+  const receiveAddresses = useReceiveAddresses(wallet)
+  const currentAddress = _.last(receiveAddresses)
```

Comment:
I'd say currentAddress should be inject as dependency the state should be agnostic to the wallet, resolve the initialCurrentAddress from the wallet and inject the into Context.